modules/c++/six.sicd/source/Utilities.cpp

```diff
@@ -130,6 +130,166 @@ Utilities::getProjectionModel(const ComplexData* data,
     }
 }
 
+six::NITFReadControl six::sicd::Utilities::getLoadedNITFReadControl(
+        const std::string& sicdPathname,
+        const std::vector<std::string>& schemaPaths)
+{
+    six::XMLControlRegistry xmlRegistry;
+    xmlRegistry.addCreator(six::DataType::COMPLEX,
+            new six::XMLControlCreatorT<
+            six::sicd::ComplexXMLControl>());
+
+    six::NITFReadControl reader;
+    reader.setXMLControlRegistry(&xmlRegistry);
+    reader.load(sicdPathname, schemaPaths); 
+    return reader;
+}
+    
+
+six::sicd::Utilities::SicdContents
+    six::sicd::Utilities::readSicd(const std::string& sicdPathname,
+             const std::vector<std::string>& schemaPaths)
+{
+    six::NITFReadControl reader = Utilities::getLoadedNITFReadControl(
+            sicdPathname,
+            schemaPaths);
+    
+    ComplexData* complexData = getComplexData(reader);
+    std::complex<float>* buffer = getWidebandData(reader, *complexData);
+    SicdContents retv;
+
+    // The reader's Container keeps ownership of the Data object
+    // so we're forced to make a copy (we can't remove the object
+    // from the container either since internally they are stored
+    // as cloneable pointers - removeData() deletes the object)
+    // TODO: shared pointers
+    retv.complexData = reinterpret_cast<ComplexData*>(complexData->clone());
+    retv.widebandData = buffer;
+    return retv;
+}
+
+ComplexData* six::sicd::Utilities::getComplexData(NITFReadControl& reader)
+{
+    // TODO: generalize this to multi-image SICDs
```

Comment:
Remove TODO - there is no such thing as a multi-image SICD.  It will always be just one image (even if under the hood it is multiple NITF image segments)